front/src/modules/scenario/components/ScenarioExplorer/ScenarioExplorer.tsx

```diff
@@ -77,6 +81,16 @@ const ScenarioExplorer = ({
     if (scenario) {
       dispatch(updateTimetableID(scenario.timetable_id));
       dispatch(updateInfraID(scenario.infra_id));
+      // We also set the stdcm environment in case we select a scenario from the stdcm interface.
+      dispatch(
+        updateStdcmEnvironment({
+          infraID: scenario.infra_id,
+          timetableID: scenario.timetable_id,
+          electricalProfileSetId: undefined,
+          workScheduleGroupId: undefined,
+          searchDatetimeWindow: undefined,
+        })
+      );
```

Comment:
It seems that we are using multiple actions (`updateTimetableID` and `updateInfraID`) to update the infrastructure and timetable IDs, and then also using the `updateStdcmEnvironment` action to update the stdcm environment with the same IDs. This can lead to redundancies and unnecessary state updates.

To simplify and improve the efficiency of the code, I propose:
- Keep the `updateStdcmEnvironment` action: Use this single action to update all necessary IDs at once.
- Remove redundant actions: Remove the `updateTimetableID` and `updateInfraID` actions to avoid redundancies.

```suggestion
      // We also set the stdcm environment in case we select a scenario from the stdcm interface.
      dispatch(
        updateStdcmEnvironment({
          infraID: scenario.infra_id,
          timetableID: scenario.timetable_id,
          electricalProfileSetId: undefined,
          workScheduleGroupId: undefined,
          searchDatetimeWindow: undefined,
        })
      );
```

Reply:
Absolutely right ! I didn't pay attention to that, thanks.